### PR TITLE
ACR Integration: Populate metadata into PSResourceInfo object

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
This PR takes the metadata json string stored in `annotations` and populates it into PSResourceInfo object.
eg:
image

A small change was made to `Publish-PSResource` as well-- it now pushes `org.opencontainers.image.title` (the package name) and `org.opencontainers.image.description` (the full .nupkg name) to `annotations`.  See [documentation on naming conventions](https://github.com/opencontainers/image-spec/blob/v1.1.0-rc4/annotations.md#pre-defined-annotation-keys). 

image

Note: the only way to retrieve `annotations` is to make a call to pull the manifest (ie GET request for a specific tag will not return layers, therefore will not return `annotations`).  See: [manifest response](https://learn.microsoft.com/en-us/rest/api/containerregistry/manifests/get?view=rest-containerregistry-2019-08-15&tabs=HTTP) vs [tag response](https://learn.microsoft.com/en-us/rest/api/containerregistry/tag/get-list?view=rest-containerregistry-2019-08-15&tabs=HTTP)

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: src/code/ACRServerAPICalls.cs [hunk 1]
```diff
@@ -218,7 +199,7 @@ public override FindResults FindName(string packageName, bool includePrerelease,
                         if (!pkgVersion.IsPrerelease || includePrerelease)
                         {
                             // Versions are always in descending order i.e 5.0.0, 3.0.0, 1.0.0 so grabbing the first match suffices
-                            latestVersionResponse.Add(new Hashtable() { { packageName, packageVersionStr } });
+                            latestVersionResponse.Add(GetACRMetadata(registry, packageName, pkgVersion, acrAccessToken, out errRecord));
```
Comment (adityapatwardhan) [inline src/code/ACRServerAPICalls.cs:202]: Should check errorRecord

File: src/code/ACRServerAPICalls.cs [hunk 2]
```diff
@@ -387,7 +351,7 @@ public override FindResults FindVersionGlobbing(string packageName, VersionRange
                                 continue;
                             }
 
-                            latestVersionResponse.Add(new Hashtable() { { packageName, packageVersionStr } });
+                            latestVersionResponse.Add(GetACRMetadata(registry, packageName, pkgVersion, acrAccessToken, out errRecord));
```
Comment (adityapatwardhan) [inline src/code/ACRServerAPICalls.cs:358]: should check errRecord

File: src/code/ACRServerAPICalls.cs [hunk 3]
```diff
@@ -453,59 +417,12 @@ public override FindResults FindVersion(string packageName, string version, Reso
...
-                {
-                    _cmdletPassedIn.WriteDebug($"'{packageName}' version parsed as '{pkgVersion}'");
-
-                    if (pkgVersion == requiredVersion)
-                    {
-                        requiredVersionResponse.Add(new Hashtable() { { packageName, packageVersionStr } });
-                    }
-                }
-            }
+                GetACRMetadata(registry, packageName, requiredVersion, acrAccessToken, out errRecord)
```
Comment (adityapatwardhan) [inline src/code/ACRServerAPICalls.cs:430]: should check errRecord

Comment (adityapatwardhan) [reply src/code/ACRServerAPICalls.cs:430]: After checking you can use AddRange

File: src/code/ACRServerAPICalls.cs [hunk 4]
```diff
@@ -722,6 +657,146 @@ internal JObject FindAcrImageTags(string registry, string repositoryName, string
...
+            if (string.IsNullOrWhiteSpace(metadataPkgName))
+            {
+                errRecord = new ErrorRecord(
+                    new InvalidOrEmptyResponse($"Response element 'packageName' is empty for package '{packageName}' in '{Repository.Name}'."),
+                    "FindNameFailure",
+                    ErrorCategory.InvalidResult,
+                    this);
+
+                return requiredVersionResponse;
+            }
```
Comment (adityapatwardhan) [inline src/code/ACRServerAPICalls.cs]: this code is repetitive. see if a method can be made to handle it.

File: src/code/ACRServerAPICalls.cs [hunk 5]
```diff
@@ -722,6 +657,146 @@ internal JObject FindAcrImageTags(string registry, string repositoryName, string
...
+
```
Comment (adityapatwardhan) [inline src/code/ACRServerAPICalls.cs]: extra line

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by anamnavi — APPROVED** [review]
LGTM

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**